cmake/compiler_flags/GNU.CXX.cmake

```diff
@@ -1,6 +1,6 @@
 if(NOT DEFINED ENV{CXXFLAGS})
     if(CMAKE_CXX_COMPILER_ID MATCHES GNU)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wno-unknown-pragmas -Wno-sign-compare -Woverloaded-virtual -Wwrite-strings -Wno-unused")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -Wall -Wno-unknown-pragmas -Wno-sign-compare -Woverloaded-virtual -Wwrite-strings -Wno-unused")
```

Comment:
C++14! Nice! I'd suggest to set this globally as a CMake property:
```
set(CMAKE_CXX_STANDARD 14)
set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
set(CMAKE_CXX_EXTENSIONS FALSE)
```
CMake will then be able to pick up later standards, it the compiler supports them.

Reply:
That would be awesome, where do I add it? The CMakeLists.txt? (It tells me not to edit it.)

Reply:
I would add a top-level file `CXXFlags.cmake` in the `compilers` folder with these contents:
```
set(CMAKE_CXX_STANDARD 14)
set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
set(CMAKE_CXX_EXTENSIONS FALSE)
set(CMAKE_EXPORT_COMPILE_COMMANDS TRUE)

include(GNU.CXX)
include(Intel.CXX)
````
You could do the same for the C flags (is there actually any C in MRCPP?) and set the standard. You'd need to modify `autocmake.yml` accordingly and run `python update.py ..` to get the changes propagated.

In XCFun we put in place something a little bit more elaborate where the validity of the flags is actually checked: https://github.com/dftlibs/xcfun/tree/master/cmake/custom/compilers but this is a bit off-topic, of course. Ping me if you need help, of course.

Reply:
Okei! I'll see if I can figure it out tomorrow! Not sure, probably not. Do you know Stig? @stigrj

Reply:
Have a look at the vampyr setup, I think it looks like the one @robertodr is suggesting. We should do the same in MRChem, by the way. I also think the C flags can be removed.

Reply:
Yes, @stigrj is right: https://github.com/MRChemSoft/vampyr/blob/master/cmake/compiler_flags/CXXFlags.cmake

Reply:
Think I fixed it.

Reply:
There's two `CXXFlags.cmake` files now :thinking: and the `-std=c++14` flag is still in the list.

Reply:
Removed the flags. You see two in cmake/compiler_fags?

Reply:
I see this https://github.com/MRChemSoft/mrcpp/pull/59/files#diff-528afcc391b811de7797116996511692 and this https://github.com/MRChemSoft/mrcpp/pull/59/files#diff-b191b6d54789eaf48c5365cb022ca5b4

Reply:
Okei, how do I fix it?

Reply:
Ah ofc sorry! For some reason I thought they both where in the same directory, and I could only see one.